typing/typecore.ml

```diff
@@ -981,102 +981,67 @@ type pattern_checking_mode =
   (** We are checking a counter-example to exhaustivity
       generated by the pattern-matching analyser. *)
 
-
 (** In [Counter_example] mode the responsibility of [type_pat] is to
-   check that a counter-example to exhaustivity is valid: it has to be
-   type-correct in a slightly different sense than for user-code, see
-   [splitting_mode] below, and also capture at least one concrete
-   value.
+   check that a counter-example to exhaustivity is valid: whether there is a
+   choice of clauses in or-patterns (including explosions of wild cards at
+   some type), which we can view as a path in a search tree, that leads
+   to a valid pattern.
+   Invalid paths are typically due to incompatible GADT equations,
+   or empty sum types.
 
    [counter_example_checking_info] contains various parameters for
    this mode of operation. *)
 and counter_example_checking_info = {
-    explode: explosion_fuel;
+    explosion_fuel: int;
     splitting_mode: splitting_mode;
-    synthetic_names: counter_example_synthetic_names;
+    constrs: (string, Types.constructor_description) Hashtbl.t;
+    labels: (string, Types.label_description) Hashtbl.t;
   }
+(** [explosion_fuel] is the depth limit on wild card explosion.
+    It is required to avoid non-termination and compilation-time slowdowns.
+    Explosion converts a wildcard to an or-pattern of the constructors of its
+    type for a variant type, or a tuple of wildcards for unary constructors,
+    records, or tuples.
```

Comment:
Can you add the docstring on the label itself instead of doing things like this?

Reply:
This is the second iteration of "where to put the documentation comments" and people are still complaining about each other's style. I think my approach (documenting the types rather than the labels) was the best, but I decided that I don't care about the details of how we do this; as long as only at most one person cares, we should be fine, and Jacques does care, so you are not allowed to care. Request denied.

Reply:
> as long as only at most one person cares, we should be fine, and Jacques does care, so you are not allowed to care.

I'm hurt.

For the record, the style used here is bad for tooling: if you don't put the docstrings at the right place, not only will it look worse in ocamldoc / odoc (which, granted, we do not use on the compiler at the moment), but you also can't hope that merlin will be able to show it to you during auto-completion (or when explicitely requesting it).

:unamused: